test/tests/functional/pbs_preemption.py

```diff
@@ -286,3 +294,51 @@ def test_vnode_resource_contention(self):
         # Since high priority job consumed only one ncpu, vnode[4]'s
         # node state should be free now
         self.server.expect(NODE, {'state': 'free'}, id='vnode[4]')
+
+    def test_host_resource_contention(self):
+        """
+        Test to make sure that preemption happens when the resource in
+        contention is host.
+        """
+        # Skip test if number of mom provided is not equal to two
+        if len(self.moms) != 2:
+            self.skipTest("test requires two MoMs as input, " +
+                          "use -p moms=<mom1>:<mom2>")
+        else:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.mom2)
+
+        a = {'resources_available.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom1)
+        a = {'resources_available.ncpus': 3}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom2)
+
+        a = {'Resource_List.select': '1:ncpus=2'}
+        j1 = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(j1)
+        j2 = Job(TEST_USER, attrs=a)
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+
+        # Stat job to check which job is running on mom1
+        pjid = jid2
+        job_stat = self.server.status(JOB, id=jid1)
+        ehost = job_stat[0]['exec_host'].partition('/')[0]
+        if ehost == self.mom1:
+            pjid = jid1
+
+        # Submit a express queue job requesting the host
+        a = {ATTR_q: 'expressq', 'Resource_List.host': self.mom1}
+        hj = Job(TEST_USER, attrs=a)
+        hjid = self.server.submit(hj)
+        self.server.expect(JOB, {'job_state': 'R'}, id=hjid)
+        self.server.expect(JOB, {'job_state': 'S'}, id=pjid)
+
+        # Submit another express queue job requesting the host,
+        # this job will stay queued
+        a = {ATTR_q: 'expressq', 'Resource_List.host': self.mom1}
+        hj2 = Job(TEST_USER, attrs=a)
+        hjid2 = self.server.submit(hj2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=hjid)
+        job_stat = self.server.status(JOB, id=hjid2)
+        print job_stat[0]['comment']
```

Comment:
This looks like a debug message.  Is it necessary?

Reply:
Ahh... a partially baked test was pushed, I've rectified it in the latest commit